sdr_repository.go

```diff
@@ -74,24 +87,53 @@ func walkSDRs(ctx context.Context, s Session) (SDRRepository, error) {
 	// duplicate it.
 	for getSDRCmd.Req.RecordID != ipmi.RecordIDLast {
 		if err := ValidateResponse(s.SendCommand(ctx, getSDRCmd)); err != nil {
-			// if we get a 0xca or 0xff, we need to implement reservations and
-			// partial reading - hopefully we'll be alright - yet to see a SDR
-			// >70 bytes long - they're specified as 64 after all.
 			return nil, err
 		}
-
-		packet := gopacket.NewPacket(getSDRCmd.Rsp.Payload, ipmi.LayerTypeSDR,
+		headerPacket := gopacket.NewPacket(getSDRCmd.Rsp.Payload, ipmi.LayerTypeSDR,
 			gopacket.DecodeOptions{
 				Lazy: true,
 				// we can't set NoCopy because we reuse getSDRCmd.Rsp
 			})
-		if packet == nil {
-			return nil, fmt.Errorf("invalid SDR: %v", getSDRCmd)
+		if headerPacket == nil {
+			return nil, fmt.Errorf("invalid SDR for record ID %d: empty packet",
+				getSDRCmd.Req.RecordID)
 		}
-		if fsrLayer := packet.Layer(ipmi.LayerTypeFullSensorRecord); fsrLayer != nil {
+		headerLayer := headerPacket.Layer(ipmi.LayerTypeSDR)
+		if headerLayer == nil {
+			return nil, fmt.Errorf("invalid SDR for record ID %d: missing SDR layer",
+				getSDRCmd.Req.RecordID)
+		}
+		header := headerLayer.(*ipmi.SDR)
+
+		if header.Type == ipmi.RecordTypeFullSensor {
+			if header.Length > 64-numBytesSDRHeader {
```

Comment:
It would be worth extracting 64 to another `const`. Can I confirm it should be `> 64-5` and not `> 64`?